OpenNMT/onmt/Optim.py

```diff
@@ -0,0 +1,59 @@
+import math
+import torch.optim as optim
+
+class Optim(object):
+
+    def _makeOptimizer(self):
+        if self.method == 'sgd':
+            self.optimizer = optim.SGD(self.params, lr=self.lr)
+        elif self.method == 'adagrad':
+            self.optimizer = optim.adagrad(self.params, lr=self.lr)
```

Comment:
Not sure if this has been fixed on another PR branch but there are typos in the optimization methods besides SGD ( they aren't capitalized)